x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransportStopTransformActionTests.java

```diff
@@ -80,20 +79,60 @@ public void testTaskStateValidationWithTransformTasks() {
             );
         final ClusterState cs = ClusterState.builder(new ClusterName("_name")).metadata(buildMetadata(pTasksBuilder.build())).build();
 
-        TransportStopTransformAction.validateTaskState(cs, Arrays.asList("non-failed-task", "failed-task"), true);
+        TransportStopTransformAction.validateTaskState(cs, List.of("non-failed-task", "failed-task"), true);
 
-        TransportStopTransformAction.validateTaskState(cs, Collections.singletonList("non-failed-task"), false);
+        TransportStopTransformAction.validateTaskState(cs, List.of("non-failed-task"), false);
 
         ClusterState.Builder csBuilderFinal = ClusterState.builder(new ClusterName("_name")).metadata(buildMetadata(pTasksBuilder.build()));
         ElasticsearchStatusException ex = expectThrows(
             ElasticsearchStatusException.class,
-            () -> TransportStopTransformAction.validateTaskState(csBuilderFinal.build(), Collections.singletonList("failed-task"), false)
+            () -> TransportStopTransformAction.validateTaskState(csBuilderFinal.build(), List.of("failed-task"), false)
         );
 
         assertThat(ex.status(), equalTo(CONFLICT));
         assertThat(
             ex.getMessage(),
-            equalTo(TransformMessages.getMessage(TransformMessages.CANNOT_STOP_FAILED_TRANSFORM, "failed-task", "task has failed"))
+            equalTo(TransformMessages.getMessage(TransformMessages.CANNOT_STOP_SINGLE_FAILED_TRANSFORM, "failed-task", "task has failed"))
+        );
+
+        // test again with two failed tasks
+        pTasksBuilder.addTask(
+            "failed-task-2",
+            TransformTaskParams.NAME,
+            new TransformTaskParams("transform-task-2", TransformConfigVersion.CURRENT, null, false),
+            new PersistentTasksCustomMetadata.Assignment("current-data-node-with-2-tasks", "")
+        )
+            .updateTaskState(
+                "failed-task-2",
+                new TransformState(
+                    TransformTaskState.FAILED,
+                    IndexerState.STOPPED,
+                    null,
+                    0L,
+                    "task has also failed",
+                    null,
+                    null,
+                    false,
+                    null
+                )
+            );
+
+        var csBuilderMultiTask = ClusterState.builder(new ClusterName("_name")).metadata(buildMetadata(pTasksBuilder.build()));
+        ex = expectThrows(
+            ElasticsearchStatusException.class,
+            () -> TransportStopTransformAction.validateTaskState(csBuilderMultiTask.build(), List.of("failed-task", "failed-task-2"), false)
+        );
+
+        assertThat(ex.status(), equalTo(CONFLICT));
+        assertThat(
+            ex.getMessage(),
+            equalTo(
+                TransformMessages.getMessage(
```

Comment:
Could you just put the full message here?
This will also verify that the substitution works correctly.